sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/SarifV1JSONImporter.java

```diff
@@ -51,93 +59,196 @@ public SerecoMetaData importResult(String data) throws IOException {
 
         return data;
     }
 
-    private String resolveDescription(ReportingDescriptor reportingDescription) {
-        return reportingDescription.getFullDescription().getText();
-    }
+    private void resolveTargetInformation(Rule rule, ResultData data, Run run) {
+        List<ReportingDescriptorRelationship> relationShips = rule.getRelationships();
+        for (ReportingDescriptorRelationship relationShip : relationShips) {
+            ReportingDescriptorReference target = relationShip.getTarget();
+            if (target == null) {
+                continue;
+            }
+            String id = target.getId();
+            if (id == null) {
+                continue;
+            }
+            ToolComponentReference toolComponent = target.getToolComponent();
+            if (toolComponent == null) {
+                continue;
+            }
+            String toolComponentName = toolComponent.getName();
+            if (toolComponentName == null) {
+                continue;
+            }
 
-    private String resolveShortName(ReportingDescriptor desc) {
-        String name = desc == null ? null : desc.getName();
-        int index = name.lastIndexOf("/");
-        if (index != -1) {
-            index++;
-            if (index < name.length() - 1) {
-                name = name.substring(index);
+            if (CWE.equalsIgnoreCase(toolComponentName)) {
+                /* cwe found, so lets look after the id */
+                data.cweId = id;
             }
         }
-        return name;
     }
 
-    private SarifSchema210 readSarifSchema210(String data) throws JsonProcessingException, JsonMappingException {
-        return mapper.readValue(data, SarifSchema210.class);
+    private String resolveDescription(Rule rule) {
+        return rule.getFullDescription().getText();
+    }
+
+    private String resolveType(Rule rule, Run run) {
+        if (rule == null) {
+            return "error:rule==null!";
+        }
+        String type = null;
+        Message shortDescription = rule.getShortDescription();
+        if (shortDescription != null) {
+            type = shortDescription.getText();
+        }
+
+        if (type == null) {
+            type = handleSpecialToolTypeResolving(rule, run);
+        }
+
+        if (type == null) {
+            /*
+             * no type identifier found, so do fallback to id, we do not use "name" becaus
+             * this is for i18n!
+             */
+            type = rule.getId();
+        }
+        return type;
+    }
+
+    private String handleSpecialToolTypeResolving(Rule rule, Run run) {
+        Tool tool = run.getTool();
+        if (tool == null) {
+            return null;
+        }
+        Driver driver = tool.getDriver();
+        if (driver == null) {
+            return null;
+        }
+        String toolName = driver.getName();
+        if (toolName == null) {
+            return null;
+        }
+        if (toolName.equalsIgnoreCase("Brakeman")) {
```

Comment:
Please don't do this. 

First, the [license of Brakeman](https://github.com/presidentbeef/brakeman/blob/main/LICENSE.md) prevents users from providing Brakeman as a service for commercial software. 

Second, we do not create implementation for a specific tool.

Reply:
As long as there are tools which not 100% implement the sarif specification we will still need to do some special treatments here. One of the reasons why we introduced our own sechub-sarif sub module here.

Some of our tests are using brakeman sarif results and inside these results the short description is not set. If we remove this special treatment we will have to change all of those tests. 

On the other hand: after reading the https://github.com/presidentbeef/brakeman/blob/main/LICENSE.md it becomes clear that this not a "real-world-scenario".

So i think you are right and I will drop the special treatment and adopt the tests.

Reply:
done: removed special brakemen treatment at all.